### ISSUE TITLE
Unfurl short form issue references

### ISSUE THREAD

**Issue by kantarjiev** [integrations/slack#885]
Before installing the github integration, our organization was in the habit of referring to issues with a shorthand of the form `repository#123`. This is ... useful, but doesn't trigger the unfurling and link preview. 

It would be great to be able to right click/hover strings of this form and get a link preview.

**Comment by miguelhughes** [comment]
yes! some sort of shorthanding would definitely be usefull. I hate to have to type (or copy paste) the whole github url every time

**Comment by brianv0** [comment]
This would be nice. I've implemented this before. It requires you to listen to a message, however, because you can only "unfurl" URLs, not the shorthand org/repo#1234 forms. Still - it'd be nice (was curious if the state of things changed - it hasn't, that's why I'm commenting)
